sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java

```diff
@@ -175,13 +190,28 @@ public boolean isActive(W window) {
   }
 
   @Override
-  public void addNew(W window) {
+  public void seenWindow(W window) {
```

Comment:
`ensureWindowExists`? `markWindowSeen` (something to make it clear that this *does* something, rather than is just an interrogation method).